src/NOTD.SC2Map/MapScript.galaxy

```diff
@@ -7791,77 +7791,85 @@ void gf_CharlieCompanyAI () {
-            }
-
-        }
-        else {
-            UnitIssueOrder(UnitGroupLoopCurrent(), Order(AbilityCommand("Sprint", 0)), c_orderQueueReplace);
-            UnitIssueOrder(UnitGroupLoopCurrent(), Order(AbilityCommand("Turbo", 0)), c_orderQueueReplace);
-            UnitIssueOrder(UnitGroupLoopCurrent(), OrderTargetingPoint(AbilityCommand("attack", 0), UnitGetPosition(UnitGroupRandomUnit(gv_marines, c_unitCountAlive))), c_orderQueueReplace);
+        if ((UnitGetType(UnitGroupLoopCurrent()) != "Marksman") && (UnitGetType(UnitGroupLoopCurrent()) != "Spectre")) {
+            UnitGroupAdd(lv_possibleEscorts, UnitGroupLoopCurrent());
         }
     }
     UnitGroupLoopEnd();
-}
-
-unit gf_CharlieCompanyAIGetFavHero (unit lp_referenceUnit) {
-    int autoE5F42EAB_n;
-    int autoE5F42EAB_i;
-
-    // Variable Declarations
-    unit lv_myFavHero;
-    int lv_localIntA;
-
-    // Variable Initialization
-    lv_localIntA = 1;
-
-    // Implementation
-    autoE5F42EAB_n = 8;
-    for (autoE5F42EAB_i = 1; autoE5F42EAB_i <= autoE5F42EAB_n; autoE5F42EAB_i += 1) {
-        if ((gv_marineHero[lv_localIntA] != null)) {
-            if ((DistanceBetweenPoints(UnitGetPosition(gv_marineHero[lv_localIntA]), UnitGetPosition(lp_referenceUnit)) <= 20.0)) {
-                lv_myFavHero = gv_marineHero[lv_localIntA];
+    if ((UnitGroupCount(lv_possibleEscorts, c_unitCountAlive) == 0)) {
+        UnitGroupAddUnitGroup(lv_possibleEscorts, gv_marines);
+    }
+    while ((gv_gameOver == false)) {
+        UnitGroupLoopBegin(gv_easyCompanyTanks);
+        for ( ; !UnitGroupLoopDone() ; UnitGroupLoopStep() ) {
+            lv_escortDist = DistanceBetweenPoints(UnitGetPosition(UnitGroupLoopCurrent()), UnitGetPosition(UnitGroupClosestToPoint(lv_possibleEscorts, UnitGetPosition(UnitGroupLoopCurrent()))));
+            if ((UnitGetType(UnitGroupLoopCurrent()) == "SiegeBreaker") && (IntToFixed(UnitGroupCount(lv_siegeTanks, c_unitCountAlive)) < (IntToFixed(UnitGroupCount(gv_easyCompanyTanks, c_unitCountAlive)) * RandomFixed(0.33, 0.67))) && (lv_escortDist <= 3.0) && (lv_escortDist >= 1.35)) {
+                AISetUnitScriptControlled(UnitGroupLoopCurrent(), true);
+                UnitGroupAdd(lv_siegeTanks, UnitGroupLoopCurrent());
+                UnitIssueOrder(UnitGroupLoopCurrent(), Order(AbilityCommand("SiegeBreakerSiege", 0)), c_orderQueueReplace);
+            }
+            else if ((UnitGetType(UnitGroupLoopCurrent()) == "SiegeBreakerSieged") && ((lv_escortDist > 3.0) || (lv_escortDist < 1.35))) {
+                AISetUnitScriptControlled(UnitGroupLoopCurrent(), true);
+                UnitGroupRemove(lv_siegeTanks, UnitGroupLoopCurrent());
+                UnitIssueOrder(UnitGroupLoopCurrent(), Order(AbilityCommand("SiegeBreakerUnsiege", 0)), c_orderQueueReplace);
+                UnitGroupAdd(lv_unescorted, UnitGroupLoopCurrent());
             }
-
         }
-        lv_localIntA += 1;
+        UnitGroupLoopEnd();
+        Wait(2.1, c_timeGame);
```

Comment:
Give enough time for tanks to un-siege before checking their position if they should continue resume escort or find a new [closer] Marine to escort.